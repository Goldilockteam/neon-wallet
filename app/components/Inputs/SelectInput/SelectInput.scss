@import '../../../styles/variables';

.selectInput {
  position: relative;

  .dropdown {
    box-sizing: border-box;
    width: 100%;
    max-height: 180px;
    overflow: auto;
    background: #f2f2f2;
    box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
    border-top: none;
    border-radius: 0 0 4px 4px;
    margin-top: -19px;
    z-index: 1;

    .dropdownItem {
      padding: 10px 20px;
      color: #000;
      font-size: 14px;
      cursor: pointer;

      &:hover {
        background: #e6e6e6;
      }

      .highlight {
        font-weight: 600;
      }
    }

    &.noSearchResults {
      padding: 10px 20px;
      color: #9e9e9e;
      text-align: center;
      font-size: 14px;
      font-style: italic;
    }
  }

  .selectInputTransparent {
    background: transparent;
  }
}
