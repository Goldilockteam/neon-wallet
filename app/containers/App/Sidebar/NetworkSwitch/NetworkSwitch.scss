@import '../../../../styles/variables';

.networkSwitch {
  color: #666;
  font-size: 0.8em;
  border-radius: 0px;

  .networkSwitchTextInput {
    background: transparent;
    border: none;
    width: 100px;
  }

  .networkSwitchTextInputWrapper {
    margin-bottom: 0;
  }

  .networkSwitchTextInputContainer {
    background-color: transparent;
    border: none;
  }

  .networkSwitchActive {
    background-color: transparent;
    border: none;
  }

  .label {
    margin-right: 4px;
    color: #999;
  }

  .dropdownItem {
    padding: 10px 20px;
    color: $black;
    font-size: 14px;
    cursor: pointer;

    &:hover {
      background: #e6e6e6;
    }

    .highlight {
      font-weight: 600;
    }
  }
}
