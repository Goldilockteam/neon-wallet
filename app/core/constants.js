// @flow
import tokenList from './tokenList.json'
import nodesMainNet from './nodes-main-net.json'
import nodesTestNet from './nodes-test-net.json'

export const NEON_WALLET_RELEASE_LINK =
  'https://github.com/CityOfZion/neon-wallet/releases'

export const DISCORD_INVITE_LINK = 'https://discordapp.com/invite/R8v48YA'
export const COZ_DONATIONS_ADDRESS = 'Adr3XjZ5QDzVJrWvzmsTTchpLRRGSzgS5A'

export const ASSETS = {
  NEO: 'NEO',
  GAS: 'GAS'
}

export const THEMES = {
  LIGHT: 'Light',
  DARK: 'Dark'
}

export const DEFAULT_THEME = THEMES.LIGHT

export const EXPLORERS = {
  NEO_SCAN: 'Neoscan',
  NEO_TRACKER: 'Neotracker',
  ANT_CHAIN: 'Antchain'
}

export const DEFAULT_EXPLORER = EXPLORERS.NEO_SCAN

export const ROUTES = {
  HOME: '/',
  DASHBOARD: '/dashboard',
  RECEIVE: '/receive',
  CONTACTS: '/contacts',
  ADD_CONTACT: '/contacts/new',
  EDIT_CONTACT: '/contacts/edit/:name',
  CREATE_WALLET: '/create',
  CREATE_WALLET_AUTHENTICATED: '/create-authenticated',
  IMPORT_WALLET: '/import',
  IMPORT_WALLET_AUTHENTICATED: '/import-authenticated',
  TOKEN_SALE: '/token-sale',
  TOKEN_SALE_CONFIRMATION: '/token-sale-confirm',
  TOKEN_SALE_FAILURE: '/token-sale-failure',
  TOKEN_SALE_SUCCESS: '/token-sale-success',
  TRANSACTION_HISTORY: '/transactions',
  SETTINGS: '/settings',
  DISPLAY_WALLET_KEYS: '/display-wallet-keys',
  DISPLAY_WALLET_KEYS_AUTHENTICATED: '/display-wallet-keys-authenticated',
  DISPLAY_WALLET_QRS: '/display-wallet-qrs',
  DISPLAY_WALLET_QRS_AUTHENTICATED: '/display-wallet-qrs-authenticated',
  WALLET_MANAGER: '/wallet-manager',
  EDIT_WALLET: '/edit-wallet/:key/:label',
  SEND: '/send/',
  SEND_ADDRESS: '/send/:address',
  ENCRYPT: '/encrypt-private-key',
  NODE_SELECT: '/node-select',
  VOTING: '/voting'
}

export const NOTIFICATION_LEVELS = {
  ERROR: 'error',
  SUCCESS: 'success',
  INFO: 'info',
  WARNING: 'warning'
}

export const NOTIFICATION_POSITIONS = {
  TOP_CENTER: 'tc',
  TOP_RIGHT: 'tr',
  TOP_LEFT: 'tl',
  BOTTOM_CENTER: 'bc',
  BOTTOM_RIGHT: 'br',
  BOTTOM_LEFT: 'bl'
}

export const BIP44_PATH = '8000002C' + '80000378' + '80000000' + '00000000' // eslint-disable-line

export const MODAL_TYPES = {
  SEND: 'SEND',
  CONFIRM: 'CONFIRM',
  TOKEN_INFO: 'TOKEN_INFO',
  TOKEN: 'TOKEN',
  ICO: 'ICO',
  ADD_CONTACT: 'ADD_CONTACT',
  RECEIVE: 'RECEIVE'
}

// export const MAIN_NETWORK_ID = '1'
export const TEST_NETWORK_ID = '2'
// export const COZ_TEST_NETWORK_ID = '3'
// export const GOLDI_NETWORK_ID = '4'

// TestNet
export const TOKENS_TEST = {
  DBC: 'b951ecbbc5fe37a9c280a76cb0ce0014827294cf',
  RPX: '5b7074e873973a6ed3708862f219a6fbf4d1c411',
  QLC: '0d821bd7b6d53f5c2b40e217c6defc8bbe896cf5',
  LOCK: 'c36a622ad044df617634460a0f68e9023c0aa4aa'
}

// MainNet
export const TOKENS = tokenList

// GoldiNet
export const TOKENS_GOLDI = {
}

export const ENDED_ICO_TOKENS = [
  'DBC',
  'RPX',
  'RHT',
  'QLC',
  'NRVE',
  'IAM',
  'ONT',
  'THOR',
  'CGE',
  'AVA',
  'SWH',
  'SWTH',
  'EFX',
  'MCT',
  'PKC'
]

export const DEFAULT_WALLET = {
  name: 'userWallet',
  version: '1.0',
  scrypt: {
    cost: 16384,
    blockSize: 8,
    parallel: 8,
    size: 64
  },
  accounts: [],
  extra: null
}

export const DEFAULT_CURRENCY_CODE = 'usd'

export const CURRENCIES = {
  aud: { symbol: '$' },
  brl: { symbol: 'R$' },
  cad: { symbol: '$' },
  chf: { symbol: 'Fr.' },
  clp: { symbol: '$' },
  cny: { symbol: '¥' },
  czk: { symbol: 'Kč' },
  dkk: { symbol: 'kr. ' },
  eur: { symbol: '€' },
  gbp: { symbol: '£' },
  hkd: { symbol: '$' },
  huf: { symbol: 'Ft ' },
  idr: { symbol: 'Rp ' },
  ils: { symbol: '₪' },
  inr: { symbol: '₹' },
  jpy: { symbol: '¥' },
  krw: { symbol: '₩' },
  mxn: { symbol: '$' },
  myr: { symbol: 'RM' },
  nok: { symbol: 'kr ' },
  nzd: { symbol: '$' },
  php: { symbol: '₱' },
  pkr: { symbol: '₨ ' },
  pln: { symbol: 'zł' },
  rub: { symbol: '₽' },
  sek: { symbol: 'kr ' },
  sgd: { symbol: 'S$' },
  thb: { symbol: '฿' },
  try: { symbol: '₺' },
  twd: { symbol: 'NT$' },
  usd: { symbol: '$' },
  zar: { symbol: 'R ' }
}

export const TOKEN_SALE_PURCHASE = 'token_sale_purchase'
export const TOKEN_SALE_CONFIRM = 'token_sale_confirm'
export const TOKEN_SALE_SUCCESS = 'token_sale_success'
export const TOKEN_SALE_FAILURE = 'token_sale_failure'

export const NODES_MAIN_NET = nodesMainNet
export const NODES_TEST_NET = nodesTestNet

export const PRICE_UNAVAILABLE = 'N/A'
