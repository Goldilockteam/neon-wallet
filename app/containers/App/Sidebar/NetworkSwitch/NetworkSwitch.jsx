// @flow
import React, { Component } from 'react'

import classNames from 'classnames'
import SelectInput from '../../../../components/Inputs/SelectInput'
import { getNetworks } from '../../../../core/networks'
import styles from './NetworkSwitch.scss'

type Props = {
  networkId: string,
  onChange: Function,
  className: string,
  networks: Array<NetworkItemType>
}

export default class NetworkSwitch extends Component<Props> {
  static defaultProps = {
    networks: getNetworks()
  }

  render() {
    const { networkId, networks, className } = this.props

    const items = networks.filter(network => network.id !== networkId)

    return (
      <div id="network" className={classNames(styles.networkSwitch, className)}>
        <SelectInput
          items={items}
          onChange={this.handleChange}
          renderItem={this.renderItem}
          activeStyles={styles.networkSwitchActive}
          textInputContainerClassName={styles.networkSwitchTextInputContainer}
          textInputClassName={styles.networkSwitchTextInput}
          value={this.getNetworkLabel()}
          readOnly
          customChangeEvent
        />
      </div>
    )
  }

  getNetworkLabel = () => {
    const { networkId, networks } = this.props
    const currentNetwork = networks.find(network => network.id === networkId)
    if (currentNetwork) {
      return currentNetwork.label
    }
    return ''
  }

  renderItem = (item: Object) => (
    <div
      key={item.id}
      onClick={this.handleChange}
      data-network={item.id}
      aria-label={item.label}
      className={styles.dropdownItem}
    >
      {item.label}
    </div>
  )

  handleChange = (event: Object) => {
    this.props.onChange(event.target.dataset.network)
  }
}
