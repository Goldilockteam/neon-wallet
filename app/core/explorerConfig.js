import { EXPLORERS } from './constants'
const { ANT_CHAIN, NEO_SCAN, NEO_TRACKER } = EXPLORERS

export default {
  [ANT_CHAIN]: {
    addressLinkStructure: 'address/info/',
    assetLinkStructure: 'asset/hash/',
    mainNetwork: 'http://antcha.in/',
    testNetwork: 'http://testnet.antcha.in/',
    trxLinkStructure: 'tx/hash/0x'
  },
  [NEO_SCAN]: {
    addressLinkStructure: 'address/',
    assetLinkStructure: 'asset/',
    mainNetwork: 'https://neoscan.io/',
    testNetwork: 'https://neoscan-testnet.io/',
    trxLinkStructure: 'transaction/'
  },
  [NEO_TRACKER]: {
    addressLinkStructure: 'address/',
    assetLinkStructure: 'asset/',
    mainNetwork: 'https://neotracker.io/',
    testNetwork: 'https://testnet.neotracker.io/',
    trxLinkStructure: 'tx/'
  }

}
