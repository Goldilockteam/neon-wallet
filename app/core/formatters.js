// @flow
import { ASSETS } from './constants'
import { toBigNumber } from './math'

export const COIN_DECIMAL_LENGTH = 8
export const SHORT_DISPLAY_DECIMAL_LENGTH = 4

type ValueType = string | number

export const formatGAS = (
  value: ValueType,
  shortDisplay: boolean = false
): string => {
  const decimals = shortDisplay
    ? SHORT_DISPLAY_DECIMAL_LENGTH
    : COIN_DECIMAL_LENGTH
  return toBigNumber(value).toFormat(decimals)
}

export const formatToRoundedShortNumber = (value: number): string => {
  if (Number.isInteger(value)) return value.toString()
  return toBigNumber(value)
    .toFormat(SHORT_DISPLAY_DECIMAL_LENGTH)
    .toString()
}

export const formatThousands = (value: ValueType): string =>
  toBigNumber(value).toFormat(0)

export const formatNEO = (value: ValueType): string =>
  toBigNumber(value).toFormat(0)

export const formatBalance = (
  symbol: SymbolType,
  balance: ValueType,
  shortDisplay: boolean = false
): string => {
  if (symbol === ASSETS.NEO) {
    return formatNEO(balance)
  }
  return formatGAS(balance, shortDisplay)
}

export const toFixedDecimals = (
  value: ValueType,
  decimals: number = COIN_DECIMAL_LENGTH
): string => toBigNumber(value).toFixed(decimals)

export const formatFiat = (value: ValueType = 0): string =>
  toBigNumber(value).toFormat(2)
