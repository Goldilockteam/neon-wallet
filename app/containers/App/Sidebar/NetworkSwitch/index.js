// @flow
import { compose } from 'recompose'
import { withActions, type Actions } from 'spunky'

import NetworkSwitch from './NetworkSwitch'
import networkActions from '../../../../actions/networkActions'
import withNetworkData from '../../../../hocs/withNetworkData'
import withAuthData from '../../../../hocs/withAuthData'
import accountActions from '../../../../actions/accountActions'
import withFilteredTokensData from '../../../../hocs/withFilteredTokensData'

const mapActionsToProps = (actions: Actions, props: Object): Object => ({
  onChange: networkId => actions.call({ networkId })
})

const mapAccountActionsToProps = (actions, props) => ({
  loadWalletData: () =>
    actions.call({
      net: props.net,
      address: props.address,
      tokens: props.tokens
    })
})

export default compose(
  withAuthData(),
  withNetworkData(),
  withActions(networkActions, mapActionsToProps),
  withFilteredTokensData(),
  withActions(accountActions, mapAccountActionsToProps)
)(NetworkSwitch)
