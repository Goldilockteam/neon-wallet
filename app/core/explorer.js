// @flow
import { openExternal } from './electron'
import explorerConfig from './explorerConfig'
import { isMainNetwork, isGoldiNetwork } from './networks'

export const buildExplorerLink = (
  networkId: string,
  explorer: ExplorerType,
  structure: string,
  suffix: string
) => {
  if (explorerConfig.hasOwnProperty(explorer)) {
    const explorerInfo = explorerConfig[explorer]

    return getExplorerBaseURL(networkId, explorer) + explorerInfo[structure] + suffix
  }

  throw new Error(`${explorer} is not a valid explorer type. Valid explorer types are: ${Object.keys(explorerConfig).join(', ')}`)
}

export const getExplorerBaseURL = (networkId: string, explorer: ExplorerType) => {
  if (explorerConfig.hasOwnProperty(explorer)) {
    const selectedExplorer = explorerConfig[explorer]

    // temporary until neon-wallet implements proper
    // per-network explorer types
    if (isGoldiNetwork(networkId))
      return 'https://neo.goldilock.com'

    return isMainNetwork(networkId)
      ? selectedExplorer.mainNetwork
      : selectedExplorer.testNetwork
  }
}

export const getExplorerTxLink = (networkId: string, explorer: ExplorerType, txId: string) => buildExplorerLink(networkId, explorer, 'trxLinkStructure', txId)

export const getExplorerAddressLink = (networkId: string, explorer: ExplorerType, address: string) => buildExplorerLink(networkId, explorer, 'addressStructure', address)

export const getExplorerAssetLink = (networkId: string, explorer: ExplorerType, assetId: string) => buildExplorerLink(networkId, explorer, 'assetLinkStructure', assetId)

export const openExplorerTx = (networkId: string, explorer: ExplorerType, txId: string) => openExternal(getExplorerTxLink(networkId, explorer, txId))

export const openExplorerAddress = (networkId: string, explorer: ExplorerType, address: string) => openExternal(getExplorerAddressLink(networkId, explorer, address))

export const openExplorerAsset = (networkId: string, explorer: ExplorerType, assetId: string) => openExternal(getExplorerAssetLink(networkId, explorer, assetId))
