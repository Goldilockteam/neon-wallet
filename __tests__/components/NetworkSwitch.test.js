import React from 'react'
import { mount } from 'enzyme'
import Select from 'react-select'

import NetworkSwitch from '../../app/containers/App/Sidebar/NetworkSwitch/NetworkSwitch'
import SelectInput from '../../app/components/Inputs/SelectInput'
import DropdownButton from '../../app/components/Inputs/SelectInput/DropdownButton'
import { MAIN_NETWORK_ID, TEST_NETWORK_ID } from '../../app/core/constants'

const NETWORKS = [
  {
    id: MAIN_NETWORK_ID,
    value: MAIN_NETWORK_ID,
    label: 'MainNet',
    network: 'MainNet'
  },
  {
    id: TEST_NETWORK_ID,
    value: TEST_NETWORK_ID,
    label: 'TestNet',
    network: 'TestNet'
  }
]

const setup = () => {
  const props = {
    value: NETWORKS[0],
    networks: NETWORKS,
    onChange: jest.fn()
  }
  const wrapper = mount(<NetworkSwitch {...props} />)

  return {
    wrapper
  }
}

describe('NetworkSwitch', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })

  test('correctly renders MainNet initially', () => {
    const { wrapper } = setup()
    const networkInput = wrapper.find(Select).instance()
    expect(networkInput.props.value.label).toEqual('MainNet')
  })

  // TODO: fix me!
  // test('switches to the correct network when chosen from the dropdown', () => {
  //   const { wrapper } = setup()
  //   const networkInput = wrapper.find(Select).instance()
  //   wrapper.find(Select).simulate('click')
  //   wrapper.find(Select).simulate('keyDown', { keyCode: 40 })
  //   wrapper.find(Select).simulate('keyDown', { keyCode: 13 })

  //   expect(networkInput.props.value.label).toEqual('TestNet')
  // })
})
