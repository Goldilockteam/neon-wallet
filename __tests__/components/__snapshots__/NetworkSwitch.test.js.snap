// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkSwitch renders without crashing 1`] = `
<NetworkSwitch
  networkId="1"
  networks={
    Array [
      Object {
        "id": "1",
        "label": "MainNet",
        "network": "MainNet",
      },
      Object {
        "id": "2",
        "label": "TestNet",
        "network": "TestNet",
      },
    ]
  }
  onChange={[MockFunction]}
>
  <div
    className="networkSwitch"
    id="network"
  >
    <SelectInput
      activeStyles="networkSwitchActive"
      customChangeEvent={true}
      getItemValue={[Function]}
      getSearchResults={[Function]}
      items={
        Array [
          Object {
            "id": "2",
            "label": "TestNet",
            "network": "TestNet",
          },
        ]
      }
      onChange={[Function]}
      onFocus={[Function]}
      readOnly={true}
      renderAfter={[Function]}
      renderItem={[Function]}
      textInputClassName="networkSwitchTextInput"
      textInputContainerClassName="networkSwitchTextInputContainer"
      value="MainNet"
    >
      <clickOutside(Dropdown)
        className="selectInput"
        onClose={[Function]}
        open={false}
        renderDropdown={[Function]}
      >
        <Dropdown
          className="selectInput"
          onClose={[Function]}
          open={false}
          renderDropdown={[Function]}
        >
          <div
            className="dropdown selectInput"
          >
            <div
              className="anchor"
            >
              <TextInput
                activeStyles="networkSwitchActive"
                className="input networkSwitchTextInputContainer"
                onChange={[Function]}
                onFocus={[Function]}
                readOnly={true}
                renderAfter={[Function]}
                textInputClassName="networkSwitchTextInput"
                type="text"
                value="MainNet"
              >
                <div
                  className="textInputContainer"
                >
                  <div
                    className="textInput input networkSwitchTextInputContainer"
                  >
                    <input
                      className="input networkSwitchTextInput"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      readOnly={true}
                      type="text"
                      value="MainNet"
                    />
                    <div
                      className="afterInput"
                    >
                      <DropdownButton
                        onToggle={[Function]}
                      >
                        <Component
                          className="dropdownButton"
                          onClick={[Function]}
                        >
                          <svg
                            className="dropdownButton"
                            onClick={[Function]}
                          />
                        </Component>
                      </DropdownButton>
                    </div>
                  </div>
                </div>
              </TextInput>
            </div>
          </div>
        </Dropdown>
      </clickOutside(Dropdown)>
    </SelectInput>
  </div>
</NetworkSwitch>
`;
